src/core/endpoints/push/remove_device.js

```diff
@@ -8,17 +8,26 @@ export function getOperation(): string {
 }
 
 export function validateParams(modules: ModulesInject, incomingParams: RemoveDeviceArgs) {
-  let { device, pushGateway } = incomingParams;
+  let { device, pushGateway, topic } = incomingParams;
   let { config } = modules;
 
   if (!device) return 'Missing Device ID (device)';
-  if (!pushGateway) return 'Missing GW Type (pushGateway: gcm or apns)';
+  if (!pushGateway) return 'Missing GW Type (pushGateway: gcm, apns or apns2)';
   if (!config.subscribeKey) return 'Missing Subscribe Key';
+
+  if (pushGateway === 'apns2') {
```

Comment:
Why not do =>
```
if ((pushGateway === 'apns2') && !topic) return 'Missing APNS2 topic';
```
?

Reply:
Fixed  in 0699a066298c3e101e0ec6ca41b41cf0bed4a40c